### ISSUE TITLE
$mdDialog resets the page scrollTop on close

### ISSUE THREAD

**Issue by herophuong** [angular/material#9881]
**Actual Behavior**:
 -  Using $mdDialog causes the page to reset its scroll position
 -  Using $mdDialog should not reset (or should restore correctly)  the page's scrollTop

**CodePen** (or steps to reproduce the issue): *
 -  [http://plnkr.co/edit/Q1UVwW0uvAgr0oRL3zBP](http://plnkr.co/edit/Q1UVwW0uvAgr0oRL3zBP)
 - Details:
   + Scroll down to see the open dialog button 
   + Open the dialog
   + Close the dialog
   + See that the page scroll position is reset

**Angular Versions**: *
 -  `Angular Version:` 1.5.8
 -  `Angular Material Version:` 1.1.1-master-fc7e9b3

**Additional Information**:
 -  `Browser Type:` Chrome 
 -  `Browser Version:`53.0.2785.143 (64-bit)
 -  `OS: *` MacOS

----
Version 1.1.1-master-fc7e9b3 seems to be the first one having this problem. Try opening/closing the dialog several times we can see that it does try to restore the scroll position of the page, albeit not reliably in all tries.
Version 1.1.1-master-e3619e6 (latest version until the publish of this), 10/10 tries of opening/closing dialog will result in the page is scrolled back to top.

**Comment by DevVersion** [comment]
I just talked to the team and we were not able to see any issues within your codepen.

> Basically it is intended to stay at the old scroll-position where you opened the dialog originally.

Please provide some more information and possibly some screenshots.

> Will reopen if we are able to reproduce this issue.

**Comment by akvaliya** [comment]
I am getting the save issue. I had purchased theme http://preview.themeforest.net/item/material-design-angular-admin-web-app/full_screen_preview/17067522?_ga=2.92927390.2019615363.1508330591-435843348.1482326494.

When i close md-dialog it scrolls page to top.

@herophuong were you able to resolve it

**Comment by herophuong** [comment]
I had to store the `scrollTop` of the page before opening the dialog and then restore it in `onRemoving` hook.

**Comment by akvaliya** [comment]
@herophuong can you provide sample code of storing **scrollTop** & **onRemoving** hook. I will help me a lot.

**Comment by herophuong** [comment]
@akvaliya Here you are:

```js
// Back-up document scroll position
var documentScrollTop = $('html, body').scrollTop();
$mdDialog.show({
    ...,
    onRemoving: function() {
        // Restore document scroll position
        $('html, body').scrollTop(documentScrollTop);
    }),
    ...
});
```

**Comment by Splaktar** [comment]
Fixed in 1.1.5. Related to https://github.com/angular/material/issues/10897.
